daemon/fqdn.go

```diff
@@ -0,0 +1,377 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	generatedLabelSource = "cilium-generated"
+	generatedLabelUUID   = "ToFQDN-UUID"
+)
+
+var (
+	namesToPoll = &pollMap{} // string -> *dnsPollTarget
+)
+
+type pollMap struct {
```

Comment:
I'm replacing this with a simple map and lock. The sync.Map proved a little hard to use and I have a race between removing a dnsPollTarget from the map when it is empty of dependent api.Rule, and another caller adding a dependent api.Rule.